sdk/prelude/src/intrinsics/felt.rs

```diff
@@ -0,0 +1,284 @@
+use core::ops::{Add, Div, Mul, Neg, Sub};
+
+#[link(wasm_import_module = "miden:prelude/intrinsics_felt")]
+extern "C" {
+    #[link_name = "from_u64_unchecked"]
+    fn extern_from_u64_unchecked(value: u64) -> Felt;
+
+    #[link_name = "as_u64"]
+    fn extern_as_u64(felt: Felt) -> u64;
+
+    #[link_name = "add"]
+    fn extern_add(a: Felt, b: Felt) -> Felt;
+
+    #[link_name = "sub"]
+    fn extern_sub(a: Felt, b: Felt) -> Felt;
+
+    #[link_name = "mul"]
+    fn extern_mul(a: Felt, b: Felt) -> Felt;
+
+    #[link_name = "div"]
+    fn extern_div(a: Felt, b: Felt) -> Felt;
+
+    #[link_name = "neg"]
+    fn extern_neg(a: Felt) -> Felt;
+
+    #[link_name = "inv"]
+    fn extern_inv(a: Felt) -> Felt;
+
+    #[link_name = "pow2"]
+    fn extern_pow2(a: Felt) -> Felt;
+
+    #[link_name = "exp"]
+    fn extern_exp(a: Felt, b: Felt) -> Felt;
+
+    #[link_name = "eq"]
+    fn extern_eq(a: Felt, b: Felt) -> i32;
+
+    #[link_name = "gt"]
+    fn extern_gt(a: Felt, b: Felt) -> i32;
+
+    #[link_name = "lt"]
+    fn extern_lt(a: Felt, b: Felt) -> i32;
+
+    #[link_name = "ge"]
+    fn extern_ge(a: Felt, b: Felt) -> i32;
+
+    #[link_name = "le"]
+    fn extern_le(a: Felt, b: Felt) -> i32;
+
+    #[link_name = "is_odd"]
+    fn extern_is_odd(a: Felt) -> i32;
+
+    #[link_name = "assert"]
+    fn extern_assert(a: Felt);
+
+    #[link_name = "assertz"]
+    fn extern_assertz(a: Felt);
+
+    #[link_name = "assert_eq"]
+    fn extern_assert_eq(a: Felt, b: Felt);
+}
+
+/// Creates a `Felt` from an integer constant checking that it is within the valid range at compile
+/// time.
+#[macro_export]
+macro_rules! felt {
+    ($value:expr) => {{
```

Comment:
If you make this a `($value:literal)` clause, you can avoid the need to do the `const VALUE..` hack entirely.

Reply:
Great hint! Thanks! Done.